sql/moz-fx-data-shared-prod/search_terms_derived/suggest_impression_sanitized_v2/query.sql

```diff
@@ -0,0 +1,47 @@
+/*
+
+This query simply copies data from a table in the restricted suggest-searches-prod
+project to shared-prod, but throws an error on empty input in order to signal an
+upstream delay or error.
+
+In order to produce the error, it includes aggregations and join steps.
+
+*/
+WITH sanitized_impressions AS (
+  SELECT
+    *
+  FROM
+    `suggest-searches-prod-a30f.sanitized.suggest_impression_sanitized_v2`
+  WHERE
+    DATE(submission_timestamp) = @submission_date
+),
+sanitized_impressions_count AS (
+  SELECT
+    COUNT(*) AS _n
+  FROM
+    sanitized_impressions
+),
+-- We perform a LEFT JOIN on TRUE as a workaround to attach the count to every
+-- row from the impressions table; the LEFT JOIN has the important property that
+-- if the input impressions partition is empty, we will still get a single row of
+-- output, which allows us to raise an error in the WHERE clause.
+validated_impressions AS (
+  SELECT
+    * EXCEPT (_n),
+  FROM
+    sanitized_impressions_count
+  LEFT JOIN
+    sanitized_impressions
+  ON
+    TRUE
+  WHERE
+    IF(
+      _n < 1,
+      ERROR("Source table is not yet populated; retry later or investigate upstream issues"),
```

Comment:
@scholtzan @relud This query introduces a new concept of checking for the source being empty and raising an error as part of the query itself. The intention here is to allow us to define the query within the bigquery-etl dag generation machinery rather than having to introduce a separate DAG in telemetry-airflow just to introduce a step that checks for content.

I'd appreciate your feedback on whether this seems reasonable and is sufficiently documented within this query file.

Reply:
Yup, this looks like a good idea 👍 
For the error message, would it make sense to mention `suggest-searches-prod-a30f.sanitized.suggest_impression_sanitized_v2` explicitly as the source table, just that it might be good to know for people debugging Airflow errors.

Reply:
I agree, it's a good approach.

VPN also does something similar to this with [sql/moz-fx-data-shared-prod/mozilla_vpn_derived/site_metrics_empty_check_v1/query.sql](https://github.com/mozilla/bigquery-etl/blob/main/sql/moz-fx-data-shared-prod/mozilla_vpn_derived/site_metrics_empty_check_v1/query.sql) and defining an explicit reference to the empty check:
https://github.com/mozilla/bigquery-etl/blob/101fe5f8e3a0c1ce4fdd3525c37df34ceeb61c51/sql/moz-fx-data-shared-prod/mozilla_vpn_derived/site_metrics_summary_v1/metadata.yaml#L13-L16

Putting it in one query, like you do here, is a nice improvement.

Reply:
Thanks for the review, @scholtzan and @relud!